packages/fxa-shared/subscriptions/types.ts

```diff
@@ -30,6 +30,8 @@ export interface Plan {
   active: boolean;
   // TODO remove the '?' here when removing the SUBSCRIPTIONS_FIRESTORE_CONFIGS_ENABLED feature flag
   configuration?: PlanConfigurationDtoT | null;
+  checkoutType?: string;
+  other?: any;
```

Comment:
In NewUserEmailForm the `plan` property was assigned type `any` which allowed previous work to attach extra properties to the plan object - this is because many of the functions to send metrics just pass in the plan object and I am assuming previous metrics work just tagged on extra data to existing inputs.

In SubscriptionCreate, the `plan` property was typed to the `Plan` type which meant I could not add a `checkoutType` to it as was already done in NewUserEmailForm. So I added `checkoutType` to the `Plan` but when I updated the type on NewUserEmailForm, there was one other extra property `other` that also needed to be added. 

We could remove it from here and change the `plan` property in NewUserEmailForm back to `any` or leave this in here. Either way, this is probably something we want to refactor in subplat3 (not having metrics stuff piggyback on the plan object).

Reply:
Yeah I saw that strange situation in `NewUserEmailForm` and was confused. `checkoutType` is not a property of a `Stripe.Plan`, and we shouldn't be modifying the `Plan` type in this way.

As a workaround for now, could you pass in `({...selectedPlan, checkoutType})` into the `Amplitude.*` functions? It's expecting `EventProperties`, so that should work.

Reply:
reverted and implemented your above suggestion